[CI] EsqlSpecIT test {csv-spec:flattened.statsByFlattenedAllRows} failing

**Build Scans:**
- [elasticsearch-intake #40007 / part5](https://gradle-enterprise.elastic.co/s/sso74e4ma5bpq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT" -Dtests.method="test {csv-spec:flattened.statsByFlattenedAllRows}" -Dtests.seed=46335E704261F395 -Dtests.locale=se-NO -Dtests.timezone=Asia/Tel_Aviv -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bcsv-spec%3Aflattened.statsByFlattenedAllRows%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Data mismatch:
row 1 column 1:a list containing
row 1 column 1:0: expected "{\"agent.id\":\"3ac5081d-9275-4cde-a2a8-a00b794a20f4\",\"agent.type\":\"filebeat\",\"host.ip\":[\"10.224.0.170\",\"fe80::4001:aff:fee0:aa\"],\"host.name\":\"devops-ci-master-blue.c.elastic-ci-prod.internal\"}" but was "{\"agent.id\":\"43baa60c-42b7-4f76-b9b3-397623f18f9c\",\"agent.type\":\"filebeat\",\"host.name\":\"infra-filebeat-mdvqm\"}"
row 2 column 1:a list containing
row 2 column 1:0: expected "{\"agent.id\":\"f767fc00-eaae-4717-a33e-19b2bd95df6d\",\"agent.type\":\"filebeat\",\"host.ip\":[\"10.224.2.157\",\"172.17.0.1\",\"fe80::4001:aff:fee0:29d\",\"fe80::42:48ff:fe1a:ed7d\",\"fe80::9c8e:44ff:fe9e:da94\"],\"host.name\":\"elasticsearch-ci-immutable-debian-9-pkg-1599049660880335683\"}" but was "{\"agent.id\":\"14903e5a-9c82-43f3-8b23-5932dd1407ac\",\"agent.type\":\"filebeat\",\"host.name\":\"infra-filebeat-z5mjq\"}"
row 3 column 1:a list containing
row 3 column 1:0: expected "{\"a
[truncated]
```

**Issue Reasons:**
- [main] 1 consecutive failures in test test {csv-spec:flattened.statsByFlattenedAllRows}
- [main] 1 consecutive failures in step part5
- [main] 1 consecutive failures in pipeline elasticsearch-intake

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -552,9 +552,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.csv.CsvFormatSpecIT
   method: test {csv-spec:csv-union-by-name.* [LOCAL]}
   issue: https://github.com/elastic/elasticsearch/issues/149482
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {csv-spec:flattened.statsByFlattenedAllRows}
-  issue: https://github.com/elastic/elasticsearch/issues/149549
 - class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
   method: test {yaml=search/30_limits/Request with negative from value in body}
   issue: https://github.com/elastic/elasticsearch/issues/149556

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/flattened.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/flattened.csv-spec
@@ -266,10 +266,13 @@ FROM flattened_otel_logs
 
 statsByFlattenedAllRows
 required_capability: group_by_flattened
+required_capability: fn_field_extract
 
 FROM flattened_otel_logs
 | STATS count = COUNT(*) BY resource.attributes
-| SORT count DESC
+| EVAL hn = field_extract(resource.attributes, "host.name")
+| SORT count DESC, hn
+| DROP hn
 ;
 
 count:long | resource.attributes:flattened
@@ -277,9 +280,9 @@ count:long | resource.attributes:flattened
 1          | "{""agent.id"":""3ac5081d-9275-4cde-a2a8-a00b794a20f4"",""agent.type"":""filebeat"",""host.ip"":[""10.224.0.170"",""fe80::4001:aff:fee0:aa""],""host.name"":""devops-ci-master-blue.c.elastic-ci-prod.internal""}"
 1          | "{""agent.id"":""f767fc00-eaae-4717-a33e-19b2bd95df6d"",""agent.type"":""filebeat"",""host.ip"":[""10.224.2.157"",""172.17.0.1"",""fe80::4001:aff:fee0:29d"",""fe80::42:48ff:fe1a:ed7d"",""fe80::9c8e:44ff:fe9e:da94""],""host.name"":""elasticsearch-ci-immutable-debian-9-pkg-1599049660880335683""}"
 1          | "{""agent.id"":""8a161b55-80d0-4ba3-afc8-31d5e15fc51c"",""agent.type"":""filebeat"",""host.ip"":[""10.224.0.109"",""172.17.0.1"",""fe80::4001:aff:fee0:6d""],""host.name"":""elasticsearch-ci-immutable-sles-15-1599243844262984968""}"
+1          | "{""host.name"":""equal-host""}"
 1          | "{""agent.id"":""631455bf-99f1-4bca-845c-7004bf7de17d"",""agent.type"":""filebeat"",""host.ip"":""10.72.1.5"",""host.name"":""filebeat-6t7v6""}"
 1          | "{""agent.id"":""8866a40d-59ec-4eef-a5c3-eee2cd9fce28"",""agent.type"":""filebeat"",""host.ip"":""10.72.3.6"",""host.name"":""filebeat-8pmfj""}"
-1          | "{""host.name"":""equal-host""}"
 1          | "{""agent.id"":""5e7a960a-4653-4973-a000-450130a32a37"",""agent.type"":""filebeat"",""host.name"":""infra-filebeat-cpcmv""}"
 1          | "{""agent.id"":""43baa60c-42b7-4f76-b9b3-397623f18f9c"",""agent.type"":""filebeat"",""host.name"":""infra-filebeat-mdvqm""}"
 1          | "{""agent.id"":""14903e5a-9c82-43f3-8b23-5932dd1407ac"",""agent.type"":""filebeat"",""host.name"":""infra-filebeat-z5mjq""}"

